test/common/router/header_formatter_test.cc

```diff
@@ -74,6 +74,22 @@ TEST_F(StreamInfoHeaderFormatterTest, TestFormatWithDownstreamLocalAddressVariab
   testFormatting("DOWNSTREAM_LOCAL_ADDRESS", "127.0.0.2:0");
 }
 
+TEST_F(StreamInfoHeaderFormatterTest, TestFormatWithDownstreamLocalPortVariable) {
+  std::array<std::tuple<Network::Address::InstanceBase*, std::string>, 3> scenarios {
```

Comment:
Genereally I haven't seen this type of thing in Envoy. When I meant three scenarios, run three individually with different args and add comment on what it tests like 

// Validate for IPv4 Address
auto address = Network::Address::InstanceConstSharedPtr{ip_address};
    EXPECT_CALL(stream_info, downstreamLocalAddress()).WillRepeatedly(ReturnRef(address)); 
    testFormatting(stream_info, "DOWNSTREAM_LOCAL_PORT", 7443)

// Validate Ipv6

// Validate for Pipe